smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/SwiftWriter.kt

```diff
@@ -157,7 +157,7 @@ class SwiftWriter(
         //
         // Also leave out the headers when JSON or the version file is being written,
         // as indicated by the file extension.
-        val isPackageManifest = listOf(PACKAGE_MANIFEST_NAME, (PACKAGE_MANIFEST_NAME + ".swift")).contains(fullPackageName)
+        val isPackageManifest = PACKAGE_MANIFEST_NAME.contains(fullPackageName)
```

Comment:
Simpler logic that achieves the same (determining if this file is a package manifest.
There is also a codegen test (see below) to ensure that manifests are written correctly.